src/credential/issuance/07-verify-and-parse-credential.ts

```diff
@@ -73,55 +73,110 @@ const parseCredentialSdJwt = (
     Logger.log(LogLevel.ERROR, "Missing claims in the credential subject");
     throw new IoWalletError("Missing claims in the credential subject"); // TODO [SIW-1268]: should not be optional
   }
+
   const attrDefinitions = credentialConfig.claims;
 
-  // the key of the attribute defintion must match the disclosure's name
+  // Validate that all attributes from the config exist in the disclosures
   const attrsNotInDisclosures = attrDefinitions.filter(
-    (definition) => !disclosures.some(([, name]) => name === definition.path[0]) // Ignore nested paths for now, see https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0-15.html#name-claims-path-pointer
+    (definition) => !disclosures.some(([, name]) => name === definition.path[0])
   );
-  if (attrsNotInDisclosures.length > 0) {
+
+  if (attrsNotInDisclosures.length > 0 && !ignoreMissingAttributes) {
     const missing = attrsNotInDisclosures.map((_) => _.path[0]).join(", ");
-    const received = disclosures.map((_) => _[1 /* name */]).join(", ");
-    if (!ignoreMissingAttributes) {
-      const message = `Some attributes are missing in the credential. Missing: [${missing}], received: [${received}]`;
-      Logger.log(LogLevel.ERROR, message);
-      throw new IoWalletError(message);
-    }
+    const received = disclosures.map((_) => _[1]).join(", ");
+    const message = `Some attributes are missing in the credential. Missing: [${missing}], received: [${received}]`;
+    Logger.log(LogLevel.ERROR, message);
+    throw new IoWalletError(message);
   }
 
-  // attributes that are defined in the issuer configuration
-  // and are present in the disclosure set
-  const definedValues = Object.fromEntries(
-    attrDefinitions
-      // retrieve the value from the disclosure set
-      .map(
-        ({ path, ...definition }) =>
-          [
-            path[0],
-            {
-              ...definition,
-              value: disclosures.find(
-                (_) => _[1 /* name */] === path[0]
-              )?.[2 /* value */],
-            },
-          ] as const
-      )
-      // add a human readable attribute name, with i18n, in the form { locale: name }
-      // example: { "it-IT": "Nome", "en-EN": "Name", "es-ES": "Nombre" }
-      .map(
-        ([attrKey, { display, ...definition }]) =>
-          [
-            attrKey,
-            {
-              ...definition,
-              name: display.reduce(
-                (names, { locale, name }) => ({ ...names, [locale]: name }),
-                {} as Record<string, string>
-              ),
-            },
-          ] as const
-      )
-  );
+  // Helper to build localized names
+  const buildName = (display: { locale: string; name: string }[]) =>
+    display.reduce(
+      (names, { locale, name }) => ({ ...names, [locale]: name }),
+      {} as Record<string, string>
+    );
+
+  // Recursive helper to build a nested object structure from a given path
+  const applyPath = (
+    target: any,
```

Comment:
Do you think it is possible to remove these `any`'s? I would also try to improve the function/variable names, I find this code a bit hard to follow.